### ISSUE TITLE
Participant Verification Status: discrepancy between top-level status and state status.

### ISSUE THREAD

**Issue by JoeArmani** [episphere/connect#969]
@FrogGirl1123 @mnataraj92 
This issue was mentioned in Gitter chat on 04/15/2024:

The participant's verification status is different in the top-level and state variables. The site has requested an update to align this participant's verification status values.

**Current values (Firestore):**
821247024: 197316935 (verified)
state.821247024:  922622075 (duplicate)
 
**Requested values (Firestore):**
821247024: 197316935 (verified)
state.821247024:  197316935 (verified)

We don't know how these values became misaligned in the first place. Could it be something in the process that marks duplicates?

**Comment by FrogGirl1123** [comment]
@KELSEYDOWLING7 could you check the data in BQ and see if it matches what Rob is reporting? 
@JoeArmani could you remind me where the top level version of verification comes from?  Sites should only be able to update verification status from either "not yet verified" or "out reach timed out" through the verification API, which I believe is the state variable.  I'm trying to figure how the heck these two versions got out of sync.

**Comment by brotzmanmj** [comment]
@mnataraj92  by any chance was the verification status changed in the SMDB interface (as an approved prod manual change)?

**Comment by KELSEYDOWLING7** [comment]
@FrogGirl1123 Excluding the many instances where d_821247024 has a value and state_d_821247024 is null, there are 11 instances where the d_821247024 and state_d_821247024 values differ. I've put the list on Box here: https://nih.app.box.com/file/1504042979451

**Comment by mnataraj92** [comment]
> @mnataraj92 by any chance was the verification status changed in the SMDB interface (as an approved prod manual change)?

Token 5a40b15f-9724-463b-b5e5-fce5e85d808a is not in the prod change log

**Comment by brotzmanmj** [comment]
It's interesting, they all involve a Duplicate and another verification status... two where the other Verification code is "Verified" and the rest where the Verification code is cannot be verified. I wonder if there is some error going on in the verification process of handling duplicates. @KELSEYDOWLING7 Are they all from one site?

**Comment by JoeArmani** [comment]
@FrogGirl1123 This variable is set on connectApp signup:
821247024: 875007964 (verification status: not yet verified)

There are ~2k participant records from three sites that use this state.821247024 variable (MF, HP, and UC).

**Comment by KELSEYDOWLING7** [comment]
@brotzmanmj  8 are from HP and 2 are from MF

**Comment by JoeArmani** [comment]
I requested additional information through Gitter and Rob (the developer) provided some helpful context:

> I discovered the issue on Monday while debugging/troubleshooting data for another recruit (Connect ID 2186415816). This recruit entered cancer comments (RcrtUP_CancerComts_v1r0) that were quite long, and was causing some errors when storing in our DB. When we call the GetParticipant API, we store a lot of the data locally and update if necessary. While investigating this issue, I noticed (by luck) that the participant in question had an excessive amount of records for the RcrtV_Verification_v1r0 value. Because there were two values (duplicate and passive) our process would keep updating our DB to the alternate value each run because it didn’t match what was in our DB.
The participant in question has two other duplicate tokens associated with them (d7ff536f-a0d4-4409-8d0f-025b3694c775 and 90065f8f-a14e-4816-a1f5-de9b0eb57cdd). They are associated with NORC issue CS0003632 where the participant updated their phone number used to log into their account and could no longer log in, so they created a new account and we needed to swap verification status’ on the accounts. This happened in January of 2023. It is an unusual situation so I’m sure that was the cause.

**Comment by brotzmanmj** [comment]
Thanks for this information. There would be a prod manual change for one or more record for this participant. I believe all of the discrepant ones will likely trace back to prod manual changes. January 2023 would be before we had the NORC interface in the SMDB to update verification status, therefore it would have a been prod manual changes that we implemented ourselves. I suggest asking Tony what he thinks since he does the prod manual changes... does he update the state version of the verification status or the non-state version?

**Comment by anthonypetersen** [comment]
Looking back on old Teams messages in one of the group chats, it was discovered in 2022 that three sites were sending verification status in as a state variable. The conclusion was reached back then that sites shouldn't be putting verification status in state as we already track in at a top-level.

@JoeArmani was able to confirm you are no longer able to send verification status in through our Submit / Update API, so I don't see any reason why we shouldn't remove all instances of state.821247024 from our database to eliminate confusion. If we correct these discrepancies and keep the data around we aren't enforcing our Data Dictionary.

**Comment by brotzmanmj** [comment]
Thank you @anthonypetersen ! Does this 'no longer able to send verification status in through our Submit / Update API' apply also the tool that NORC uses to update verification status in the SMDB?

I agree with removing all instances of the state variable once @mnataraj92 is certain the handful of ~12 that are discrepant have the correct verification status in the non-state version of the variable. @FrogGirl1123 do you agree?

**Comment by anthonypetersen** [comment]
The tool NORC uses updates verification status in the correct top-level position, so we're good there.

**Comment by mnataraj92** [comment]
d_821247024 is correct for these individuals; I will add it in the prod change log to remove all instances of the state variable

**Comment by JoeArmani** [comment]
Thanks. I notified the site that the requested update is scheduled.

**Comment by brotzmanmj** [comment]
@FrogGirl1123 do you agree with Tony's suggestion to remove all instances of state.821247024 from our database to eliminate confusion? There are 2084 of them.

**Comment by FrogGirl1123** [comment]
@brotzmanmj yes, I'm in agreement. This way no one will accidently refer to the wrong version of 821247024 again.

**Comment by robertsamm** [comment]
> [@FrogGirl1123](https://github.com/FrogGirl1123) do you agree with Tony's suggestion to remove all instances of state.821247024 from our database to eliminate confusion? There are 2084 of them.

 @anthonypetersen do you know if this has been completed and can be closed out?
